lib/galaxy/webapps/galaxy/api/library_contents.py

```diff
@@ -209,9 +212,16 @@ def create( self, trans, library_id, payload, **kwd ):
 
         # are we copying an HDA to the library folder?
         #   we'll need the id and any message to attach, then branch to that private function
-        from_hda_id, ldda_message = ( payload.pop( 'from_hda_id', None ), payload.pop( 'ldda_message', '' ) )
-        if create_type == 'file' and from_hda_id:
-            return self._copy_hda_to_library_folder( trans, from_hda_id, library_id, real_folder_id, ldda_message )
+        from_hda_id, from_hdca_id, ldda_message = ( payload.pop( 'from_hda_id', None ), payload.pop( 'from_hdca_id', None ), payload.pop( 'ldda_message', '' ) )
+        log.debug(payload)
+        if create_type == 'file':
+            rval = []
+            if from_hda_id:
+                rval.append(self._copy_hda_to_library_folder( trans, self.hda_manager, self.decode_id(from_hda_id), real_folder_id, ldda_message ))
```

Comment:
This will break BioBlend.objects (and maybe other user scripts), which expects the return value to be a `dict` when copying a single file using `from_hda_id`, see https://travis-ci.org/nsoranzo/bioblend/jobs/213674739#L1484 .

Not sure if it is better to restore the previous behaviour or workaround it in BioBlend.